src/group_theory/submonoid/operations.lean

```diff
@@ -93,13 +93,64 @@ def add_submonoid.of_submonoid {M : Type*} [add_zero_class M] (S : submonoid (mu
   zero_mem' := S.one_mem',
   add_mem' := S.mul_mem' }
 
+lemma submonoid.to_add_submonoid_coe {M : Type*} [mul_one_class M] (S : submonoid M) :
+  (S.to_add_submonoid : set (additive M)) = additive.of_mul '' S :=
+eq.symm (set.image_id S)
+
+lemma add_submonoid.to_submonoid_coe {M : Type*} [add_zero_class M] (S : add_submonoid M) :
+  (S.to_submonoid : set (multiplicative M)) = multiplicative.of_add '' S :=
+eq.symm (set.image_id S)
+
+lemma submonoid.to_add_submonoid_le_iff {M : Type*} [mul_one_class M] (S₁ S₂ : submonoid M) :
+  S₁ ≤ S₂ ↔ S₁.to_add_submonoid ≤ S₂.to_add_submonoid :=
+iff.rfl
+
+lemma add_submonoid.to_submonoid_le_iff {M : Type*} [add_zero_class M] (S₁ S₂ : add_submonoid M) :
+  S₁ ≤ S₂ ↔ S₁.to_submonoid ≤ S₂.to_submonoid :=
+iff.rfl
```

Comment:
I think these might make more sense to state in terms of `mono`, like `subring.to_submonoid_mono` and `subring.to_submonoid_strict_mono`. Could you add those lemmas instead of these?